python/gcp_workspace_table_to_dataset_ingest.py

```diff
@@ -144,12 +152,31 @@ def get_args() -> argparse.Namespace:
         logging.warning("Dataset is not selfHosted. Cannot check if files have already been ingested and use UUIDs")
         check_if_files_already_ingested = False
 
+    if trunc_and_reload:
+        logging.info(
+            "Requested trunc and reload, all Terra tables marked for ingestion will be soft-deleted if they exist in "
+            "the target TDR dataset"
+        )
+        for terra_table in terra_tables:
+            try:
+                tdr.soft_delete_all_table_entries(dataset_id=dataset_id, table_name=terra_table)
+            except HTTPError:
+                logging.warning(f"Table with name '{terra_table}' does not exist in TDR dataset. Skipping soft-delete")
```

Comment:
Could you put the soft delete after 
```
        SetUpTDRTables(
            tdr=tdr,
            dataset_id=dataset_id,
            table_info_dict=table_info_dict,
            all_fields_non_required=all_fields_non_required,
            force_disparate_rows_to_string=force_disparate_rows_to_string,
        ).run()
```
and then you wouldn't need to worry about if the table exists and catching the failure?

Reply:
mm ya good point, thanks! updated